apps/teams-test-app/src/components/AuthenticationAPIs.tsx

```diff
@@ -63,9 +100,15 @@ const Authenticate = (): React.ReactElement =>
           throw new Error('url is required');
         }
       },
-      submit: async authParams => {
-        const token = await authentication.authenticate(authParams);
-        return 'Success: ' + token;
+      submit: {
+        withPromise: async authParams => {
+          const token = await authentication.authenticate(authParams);
+          return 'Success: ' + token;
+        },
+        withCallback: authParams => {
+          authentication.authenticate(authParams);
```

Comment:
shouldn't you need to set `authParams.successCallback` and `authParams.failureCallback` before passing them to `authentication.authenticate`?

Reply:
Good catch, you are correct.